src/realm/set.hpp

```diff
@@ -72,6 +72,15 @@ class Set final : public CollectionBaseImpl<SetBase, Set<T>> {
         if (ndx >= current_size) {
             throw std::out_of_range("Index out of range");
         }
+
+        // proxy out the mixed value for links, we need to know if the link is valid or not (not necesseraly must it
+        // be null)
+        if constexpr (std::is_same<T, Mixed>::value) {
+            Mixed mixed_link_value = m_tree->get(ndx);
+            if (mixed_link_value.is_null())
```

Comment:
did you intend to check `mixed_link_value.is_unresolved_link()`?

Reply:
Yes, I wanted to call ```is_unresolved_link()```